ccm_web/server/src/api/dtos/api.get.sections.admin.dto.ts

```diff
@@ -0,0 +1,29 @@
+import { Transform } from 'class-transformer'
+import { IsInt, IsNotEmpty, IsOptional, MaxLength, MinLength } from 'class-validator'
+import { ApiProperty } from '@nestjs/swagger'
+
+export class GetSectionsAdminQueryDto {
+  @ApiProperty({ type: Number })
+  @IsInt()
+  @IsNotEmpty()
+  @Transform(({ value }) => Number.parseInt(value))
+  term_id: number
+
+  @ApiProperty({ type: String, required: false })
+  @IsOptional()
+  @MinLength(3)
+  @MaxLength(255)
+  instructor?: string
```

Comment:
Anytime search input is < 3 characters in instructor or course name the response will look like below. Probably a better variable name is `instructor_name` message saying `instructor_name` must be longer than or equal to 3 characters
```
{
  "statusCode": 400,
  "message": [
    "instructor must be longer than or equal to 3 characters"
  ],
  "error": "Bad Request"
}
```

Reply:
An Observation from above comment w.r.t error response object. 
I don't think FE is capturing this object since `APIErrorData` `message ` is string instead of array. I think if something needs to change, it is beyond the scope of the issues
```
export interface APIErrorData {
  statusCode: number
  message?: string
  redirect?: true
  error?: string
  errors: APIErrorPayload[]
}
```

Reply:
Yeah, the typing needs to be improved here for sure in a couple ways. Probably it should be something like:

```
export interface APIErrorData {
  statusCode: number
  error?: string
}

export interface ApplicationAPIErrorData extends APIErrorData {
  message?: string | string[]
  redirect?: string
}

export interface CanvasAPIErrorData extends APIErrorData {
  errors: APIErrorPayload[]
}
```

with some maybe some type guards. This is indeed a separate issue.

Reply:
https://github.com/tl-its-umich-edu/canvas-course-manager-next/issues/206